native/cocos/math/Vec3.cpp

```diff
@@ -192,15 +222,63 @@ void Vec3::transformMat3(const Vec3 &v, const Mat3 &m) {
     z = ix * m.m[2] + iy * m.m[5] + iz * m.m[8];
 }
 
-void Vec3::transformMat4(const Vec3 &v, const Mat4 &m) {
+void Vec3::transformMat4Neon(const Vec3 &v, const Mat4 &m) {
+#if defined(USE_NEON64) || defined(USE_NEON32) || defined(INCLUDE_NEON32)
+    alignas(16) float tmpV0[4];
+
+    auto row0 = enoki::load<SimdVec4>(&m.m[0]);
+    auto row1 = enoki::load<SimdVec4>(&m.m[4]);
+    auto row2 = enoki::load<SimdVec4>(&m.m[8]);
+    auto row3 = enoki::load<SimdVec4>(&m.m[12]);
+
+    row0 *= v.x;
+    row1 *= v.y;
+    row2 *= v.z;
+
+    row0 = row0 + row1 + row2 + row3;
+    enoki::store(tmpV0, row0);
+    float rhw;
+    // FIXME: use std::numeric_limits<float>::epsilon() instead ?
+    static constexpr float FLT_PRECISION = 0.000001F;
+    if (CC_PREDICT_TRUE(std::abs(tmpV0[3]) > FLT_PRECISION)) {
```

Comment:
Can add a `isGreaterThanF()`? As there is `isEqualF()`.

Reply:
For optimization reason.
It originally invokes `math::isNotEqualF`.
```c++
float rhw = math::isNotEqualF(tmp[3], 0.0F) ? 1 / tmp[3] : 1; // Pass extra number 0.0F to isNotEqualF, it's not needed.
```
I extend it here just because I don't wanna subtract `0` here.
```c++
inline bool isNotEqualF(float lhs, float rhs, float precision = 0.000001F) { // rhs is 0
    return (std::abs(lhs - rhs) > precision);
}
```

Reply:
Yes, adding a isGreaterThanF function is better.

Reply:
Add math::isNotZeroF, please review again.